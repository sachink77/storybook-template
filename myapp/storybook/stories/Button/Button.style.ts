import {StyleSheet} from 'react-native';

 export default StyleSheet.create({
   button: {
     display: 'flex',
     width:75,
     backgroundColor: 'green',
     borderRadius: 15,
     alignItems: 'center',
     justifyContent: 'center',
     padding:10,
   },
 });