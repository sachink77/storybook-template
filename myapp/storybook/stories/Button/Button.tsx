import React from 'react';
import {TouchableHighlight} from 'react-native';
import {Text} from 'react-native';
import styles from './Button.style';

interface IButtonProps {
  onPress: () => void;
}

export const Button = ({onPress}: IButtonProps) => {
  return (
    <TouchableHighlight style={styles.button} onPress={onPress}>
      <Text>hi</Text>
    </TouchableHighlight>
  );
};
