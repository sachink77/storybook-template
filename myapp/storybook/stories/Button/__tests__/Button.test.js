import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from '../Button';
import {noop} from 'lodash';
import {TouchableHighlight} from 'react-native';

describe('Button', () => {
  it('should render properly', () => {
    const component = renderer.create(<Button onPress={noop} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should action callback on click', () => {
    const action = jest.fn();
    const component = renderer.create(<Button onPress={action} />);

    //press on button
    const button = component.root.findByType(TouchableHighlight);
    expect(button).toBeDefined();
    button.props.onPress();

    //expect action to be called once
    expect(action).toHaveBeenCalledTimes(1);
  });
});
