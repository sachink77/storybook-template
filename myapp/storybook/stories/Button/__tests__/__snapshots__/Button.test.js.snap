// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": 15,
      "display": "flex",
      "justifyContent": "center",
      "padding": 10,
      "width": 75,
    }
  }
>
  <Text>
    hi
  </Text>
</View>
`;
